packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -145,7 +149,7 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 	}
 
 	public getWorkspace<TSchema extends PresenceStatesSchema>(
-		internalWorkspaceAddress: string,
+		internalWorkspaceAddress: InternalWorkspaceAddress,
```

Comment:
It looks like you are safeguarding "system" name? Can app developers retrieve any workspace? Including system? Im trying to interpret "internal" here.

Reply:
There are `WorkspaceAddress`s (public) and `InternalWorkspaceAddress`s (what is actually stored from the original customer `WorkspaceAddress`). That later address is only used internally.

Reply:
In the example, I have not seen explicit call by app developer to `getWorkspace` (or maybe I missed it). It's usually just a workspace address id that helps locate value manager. Does app developer ever need to explicitly retreive workspace object?

Reply:
"Workspace" is the concept name. In API there are two types of workspaces that customers work with: 1) states workspace and 2) notifications workspace. Those are accessed by `getStates` and `getNotifications` respectively.
In the one in-repo example we have, `getStates` is found in examples/service-clients/azure-client/external-controller/src/presence.ts. `buildDicePresence` returns `DicePresence` which is the states workspace that the example works with.
All of the test code for value managers in package uses `getStates` or `getNotifications`.